ql/src/java/org/apache/hadoop/hive/ql/metadata/PartitionIterable.java

```diff
@@ -150,18 +157,36 @@ public PartitionIterable(Hive db, Table table, Map<String, String> partialPartit
     this(db, table, partialPartitionSpec, batchSize, false);
   }
 
+  public PartitionIterable(Hive db, Table table, Map<String, String> partialPartitionSpec,
+                           int batchSize, boolean isAuthRequired, String userName,
+                           List<String> groupNames) throws HiveException {
+    this(db, table, partialPartitionSpec, batchSize, false, isAuthRequired, userName, groupNames);
+  }
+
+
   /**
    * Primary constructor that fetches all partitions in a given table, given
    * a Hive object and a table object, and a partial partition spec.
    */
   public PartitionIterable(Hive db, Table table, Map<String, String> partialPartitionSpec,
-                           int batchSize, boolean getColStats) throws HiveException {
+       int batchSize, boolean getColStats) throws HiveException {
+    this(db, table, partialPartitionSpec, batchSize, getColStats, false, null, null);
+  }
+
+  public PartitionIterable(Hive db, Table table, Map<String, String> partialPartitionSpec,
+       int batchSize, boolean getColStats, boolean isAuthRequired, String userName,
+       List<String> groupNames) throws HiveException {
     this.currType = Type.LAZY_FETCH_PARTITIONS;
     this.db = db;
     this.table = table;
     this.partialPartitionSpec = partialPartitionSpec;
     this.batchSize = batchSize;
     this.getColStats = getColStats;
+    this.isAuthRequired = isAuthRequired;
+    if (isAuthRequired) {
```

Comment:
Is userName/groupName set without isAuthRequired ever?

Reply:
made the necessary changes